### DISCUSSION TITLE
OpenApi resources

### DISCUSSION THREAD

**Discussion by Neirda24** [symfony/symfony#42452] [Q&A]
Hi. I often write openapi doc when using the symfony framework.
When using the validator component and the ConstraintViolationList normalizer, it appears to me that we have to re-write everything about this schema on each new project. 

What if symfony.com could host some ressources that we could include in our openapi docs ? (see https://swagger.io/docs/specification/using-ref/ `URL reference`).
It could then look like this :
```yaml
responses:
  '400':
    description: The response
    schema: 
      $ref: 'https://symfony.com/.openapi/master/ContraintViolationList.yaml'
```

What do you think ?

**Comment by lyrixx** [comment]
This has already been discused 
* https://github.com/symfony/symfony-docs/issues/10001
* https://github.com/symfony/symfony/pull/22150#issuecomment-291250171
* https://github.com/symfony/symfony/pull/15154

I'm 👍🏼 with that :)

**Reply by Neirda24** [reply]
On what you linked I do not see exactly what I meant. It is suggested to have a hosted documentation on codes. What I suggest is to have openapi format of those violations. Since having all of the codes on all responses wouldn't make sense I would document just the structure of the violation list (in json or other format natively supported by symfony). Regarding the code list and it's meaning specifically I don't know... maybe the same way but each in a different namespace : `https://symfony.com/.openapi/master/Violations/Length.yaml` for example

**Comment by Tobion** [comment]
> it appears to me that we have to re-write everything about this schema on each new project.

Why not create a small re-usable library that people can install and then reference the schema? If this library becomes very popular we can think about including something like this in core.
